### PR TITLE
OIDC support

### PR DESCRIPTION
**Pull request by wpjunior**

### CODE REVIEW

**Review by nettoclaudio — CHANGES_REQUESTED** [review]
It looks good to me at all... I requested some mostly cosmetic changes. 💅

By the way, clever solution to support multi-auth schemes simultaneously. Gonna be important to support switching from one authorization server to another. 🥇

File: auth/oidc/oidc.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		return nil, err
+	}
+
+	if s.jwksURL == "" {
+		return nil, errNoJWKSURLS
+	}
+
+	identity := &extendedClaims{}
+
+	token = strings.TrimPrefix(token, "Bearer ")
```
[the remark this answers is not in the document]
Comment (wpjunior) [reply auth/oidc/oidc.go:81]: not yet, do you thing that is important?

[the remark this answers is not in the document]
Comment (nettoclaudio) [reply auth/oidc/oidc.go:81]: I think so... it's the way default Tsuru client send it today, look: https://github.com/tsuru/tsuru/blob/f41fb455f8c3903fb947d7bc5e190448a511ac5f/cmd/client.go#L87

**Review by nettoclaudio — APPROVED** [review]
Go for it.

**Pull request merged**